tike/ptycho.py

```diff
@@ -365,7 +365,8 @@ def grad(
         # Go far-plane
         farplane = np.fft.fft2(nearplane_pad)
         # Replace the amplitude with the measured amplitude.
-        farplane = (np.sqrt(data) * (farplane.real + 1j * farplane.imag)
+        farplane[farplane == 0] = 1  # no division by zero allowed
+        farplane = (np.sqrt(data) * farplane
```

Comment:
Is this an element-wise multiplication?

Reply:
Yes. For NumPy arrays, you have to use `@` a.k.a `np.matmul` to do matrix multiplication.